src/ripple/rpc/handlers/AMMInfo.cpp

```diff
@@ -78,32 +78,54 @@ doAMMInfo(RPC::JsonContext& context)
 
     Issue issue1;
     Issue issue2;
+    uint256 ammID;
+    bool const isAssets =
+        params.isMember(jss::asset) && params.isMember(jss::asset2);
+    bool const isAMMAccount = params.isMember(jss::amm_account);
```

Comment:
If the variables are made optional we can remove the `isAssets` and `isAMMAccount` and just check if the optional is set.